hazelcast/serialization/service.py

```diff
@@ -1,67 +1,226 @@
+            config,
+            self._portable_serializer,
+            self._data_serializer,
+            self._compact_stream_serializer,
+            self._null_serializer,
+            self._python_serializer,
+            self._global_serializer,
+        )
+
         self._register_constant_serializers()
 
         # Register Custom Serializers
         for _type, custom_serializer in config.custom_serializers.items():
             self._registry.safe_register_serializer(custom_serializer(), _type)
 
-        # Register Global Serializer
-        global_serializer = config.global_serializer
-        if global_serializer:
-            self._registry._global_serializer = global_serializer()
+    def to_data(self, obj, partitioning_strategy=None):
+        """Serialize the input object into byte array representation
+
+        Args:
+            obj: Input object
+            partitioning_strategy (function): Function in the form of ``lambda key: partitioning_key``.
+
+        Returns:
+            hazelcast.serialization.data.Data: Data object
+        """
+        if obj is None:
+            return None
+
+        if isinstance(obj, Data):
+            return obj
+
+        out = _ObjectDataOutput(self._output_buffer_size, self, self._is_big_endian)
+        try:
+            serializer = self._registry.serializer_for(obj)
+            partitioning_hash = self._calculate_partitioning_hash(obj, partitioning_strategy)
+
+            out.write_int_big_endian(partitioning_hash)
+            out.write_int_big_endian(serializer.get_type_id())
+            serializer.write(out, obj)
+            return Data(out.to_byte_array())
+        except:
+            handle_exception(sys.exc_info()[1], sys.exc_info()[2])
+
+    def to_object(self, data):
+        """Deserialize input data
+
+        Args:
+            data (hazelcast.serialization.data.Data): Serialized input Data object
+
+        Returns:
+            any: Deserialized object
+        """
+        if not isinstance(data, Data):
+            return data
+
+        if data.data_size() == 0 and data.get_type() == CONSTANT_TYPE_NULL:
+            return None
+
+        inp = _ObjectDataInput(data.buffer, DATA_OFFSET, self, self._is_big_endian)
+        try:
+            type_id = data.get_type()
+            serializer = self._registry.serializer_by_type_id(type_id)
+            if serializer is None:
+                if self._active:
+                    raise HazelcastSerializationError("Missing Serializer for type-id:%s" % type_id)
+                else:
+                    raise HazelcastInstanceNotActiveError()
+            return serializer.read(inp)
+        except:
+            handle_exception(sys.exc_info()[1], sys.exc_info()[2])
+
+    def write_object(self, out, obj):
+        if isinstance(obj, Data):
+            raise HazelcastSerializationError(
+                "Cannot write a Data instance! Use write_data(out, data) instead."
+            )
```

Comment:
Yes, apparently we don't want that. But let me remove this, as data is not part of the public API since 4.0, and there is no write_data method anymore